### PR TITLE
Fixes 3790: Add template details

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary

- Adds template details pages

## Testing steps

- Create a template, click on the the created template from the template list table, view associated information
- Read vs Write permissions implemented fully.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3790

**Comment by xbhouse** [comment]
it seems like the Advisories tab routing isn't working correctly. clicking the Advisories tab in the Template Details page always routes back to the Packages tab

**Comment by xbhouse** [comment]
Screenshot 2024-06-18 at 11 21 17 AM

name column alignment seems a little off, maybe there's some extra padding?

File: src/Pages/Templates/TemplateDetails/TemplateDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+              <Flex direction={{ default: 'column' }}>
+                <DetailItem title='Description:' value={data?.description} />
+                <DetailItem
+                  title='Includes content up to:'
+                  value={data?.date ? formatDateDDMMMYYYY(data.date) : ''}
+                />
+              </Flex>
+              <Flex direction={{ default: 'column' }}>
+                <DetailItem title='Created by:' value={data?.created_by} />
+                <DetailItem title='Created:' value={data?.created_at} />
```
Comment (xbhouse) [inline src/Pages/Templates/TemplateDetails/TemplateDetails.tsx:121]: i think there should be some conditional date formatting here similar to what you have for `date` and `updated_at`

Comment (Andrewgdewar) [reply src/Pages/Templates/TemplateDetails/TemplateDetails.tsx:121]: Good shout, checking.

**Comment by Andrewgdewar** [comment]
> it seems like the Advisories tab routing isn't working correctly. clicking the Advisories tab in the Template Details page always routes back to the Packages tab
> 
> also clicking Edit in the actions dropdown from the Template Details page routes back to repositories. something fishy going on with routing 🤔

Did you make sure to nvm use, and npm install?

**Comment by xbhouse** [comment]
@Andrewgdewar yep i did. are you not seeing that behavior?

**Comment by xbhouse** [comment]
if i print out the pathname when clicking the Advisories tab i see an extra slash but i haven't figured out where it's coming from yet: `/insights/content/templates/9acc5327-962d-4af0-94b1-741486cce0af/details//content/advisories`

File: src/Pages/Templates/TemplateDetails/components/TemplateActionDropdown.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    useDeleteTemplateItemMutate(queryClient);
+
+  const onToggleClick = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const onSelect = (_, value: string | number | undefined) => {
+    switch (value) {
+      case 'edit':
+        navigate(`${baseRoute}/${uuid}/edit`);
```
Comment (xbhouse) [inline src/Pages/Templates/TemplateDetails/components/TemplateActionDropdown.tsx:36]: same thing here as the tab routing, baseRoute ends with a `/` so the extra one between baseRoute and uuid breaks things

**Comment by xbhouse** [comment]
if i'm at the top of the page on the Advisories tab, clicking the dropdown filter makes the page jump and i have to scroll back up to look at the dropdown. it doesn't happen if i'm not at the top of the page when clicking. this could have been an issue before and not related to this PR:

https://github.com/content-services/content-sources-frontend/assets/28575816/3bb0e738-1b9b-4648-a2da-4b155def6fee

**Comment by Andrewgdewar** [comment]
> if i'm at the top of the page on the Advisories tab, clicking the dropdown filter makes the page jump and i have to scroll back up to look at the dropdown. it doesn't happen if i'm not at the top of the page when clicking. this could have been an issue before and not related to this PR:
> 
>  Screen.Recording.2024-06-25.at.3.28.40.PM.mov

Bizarre... 
I'll take a look at this.
Are you on Chrome?

**Comment by xbhouse** [comment]
> > if i'm at the top of the page on the Advisories tab, clicking the dropdown filter makes the page jump and i have to scroll back up to look at the dropdown. it doesn't happen if i'm not at the top of the page when clicking. this could have been an issue before and not related to this PR:
> > Screen.Recording.2024-06-25.at.3.28.40.PM.mov
> 
> Bizarre... I'll take a look at this. Are you on Chrome?

yep, and looks like the same thing happens on firefox

**Comment by Andrewgdewar** [comment]
Thanks for the debug help @xbhouse, I was able to resolve this by removing a trailing forward slash in the "baseRoute" in a couple places.

**Review by xbhouse — APPROVED** [review]
looks good!! ack! :tada:

**Comment by swadeley** [comment]
Hi

lgtm

image

**Pull request merged**